test/system/components/student/editor_test.rb

```diff
@@ -1,106 +1,119 @@
-          submission: Submission.last,
-          status: "error",
-          message: "Undefined local variable",
-          ops_status: 200,
-          tests: []
-        Submission::TestRunsChannel.broadcast!(test_run)
-
-        assert_text "Status: error"
-        assert_text "Undefined local variable"
+        use_capybara_host do
+          visit test_components_student_editor_path(solution_id: solution.id)
+          click_on "Run tests"
+          wait_for_submission
+          2.times { wait_for_websockets }
+          test_run = create :submission_test_run,
+            submission: Submission.last,
+            status: "error",
+            message: "Undefined local variable",
+            ops_status: 200,
+            tests: []
+          Submission::TestRunsChannel.broadcast!(test_run)
+
+          assert_text "Status: error"
+          assert_text "Undefined local variable"
+        end
       end
 
       test "user runs tests and an ops error happens" do
         user = create :user
         create :user_auth_token, user: user
         solution = create :concept_solution, user: user
 
-        visit test_components_student_editor_path(solution_id: solution.id)
-        click_on "Run tests"
-        wait_for_submission
-        2.times { wait_for_websockets }
-        test_run = create :submission_test_run,
-          submission: Submission.last,
-          status: "error",
-          message: "Can't run the tests",
-          ops_status: 400,
-          tests: []
-        Submission::TestRunsChannel.broadcast!(test_run)
-
-        assert_text "Status: ops_error"
-        assert_text "Can't run the tests"
+        use_capybara_host do
+          visit test_components_student_editor_path(solution_id: solution.id)
+          click_on "Run tests"
+          wait_for_submission
+          2.times { wait_for_websockets }
+          test_run = create :submission_test_run,
+            submission: Submission.last,
+            status: "error",
+            message: "Can't run the tests",
+            ops_status: 400,
+            tests: []
+          Submission::TestRunsChannel.broadcast!(test_run)
+
+          assert_text "Status: ops_error"
+          assert_text "Can't run the tests"
+        end
       end
 
       test "user runs tests and cancels" do
         user = create :user
         create :user_auth_token, user: user
         solution = create :concept_solution, user: user
 
-        visit test_components_student_editor_path(solution_id: solution.id)
-        click_on "Run tests"
-        wait_for_submission
-        click_on "Cancel"
+        use_capybara_host do
+          visit test_components_student_editor_path(solution_id: solution.id)
+          click_on "Run tests"
+          wait_for_submission
+          click_on "Cancel"
 
-        assert_text "Status: cancelled"
+          assert_text "Status: cancelled"
+        end
       end
 
       private
       def wait_for_submission
```

Comment:
This method name implies that it is waiting for something. With the removal of the `wait` it no longer waits. Maybe inline this method, or else add a `wait_for_cancellation` method to make the 

```
click_on "Run tests"
wait_for_submission

click_on "Cancel"
assert_text "Status: cancelled"
```

be slightly more consistent.

Reply:
Will do :)

Reply:
I thought about this again, and I think it's fine. When capybara asserts things, it waits for the result implicitly. I wouldn't also want to inline this as it takes the form of `assert_text` which sort of means like it's an assertion. But this is the only way for us to ensure that we wait for the API request to get back.

Reply:
That's a perfectly fine reason. Should probably document it inside at `wait_for_submission`, to explain WHY.